components/kserve/kserve_config_handler.go

```diff
@@ -0,0 +1,163 @@
+package kserve
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+)
+
+const (
+	KserveConfigMapName string = "inferenceservice-config"
+)
+
+func (k *Kserve) setupKserveConfigAndDependencies(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec) error {
+	// as long as Kserve.Serving is not 'Removed', we will setup the dependencies
+
+	switch k.Serving.ManagementState {
+	case operatorv1.Managed:
+		// check on dependent operators if all installed in cluster
+		dependOpsErrors := checkDependentOperators(cli).ErrorOrNil()
+		if dependOpsErrors != nil {
+			return dependOpsErrors
+		}
+
+		if err := k.configureServerless(dscispec); err != nil {
+			return err
+		}
+
+		if k.DefaultDeploymentMode == "" {
+			// if the default mode is empty in the DSC, assume mode is "Serverless" since k.Serving is Managed
+			if err := k.setDefaultDeploymentMode(ctx, cli, dscispec, Serverless); err != nil {
+				return err
+			}
+		} else {
+			// if the default mode is explicitly specified, respect that
+			if err := k.setDefaultDeploymentMode(ctx, cli, dscispec, k.DefaultDeploymentMode); err != nil {
+				return err
+			}
+		}
+
+	case operatorv1.Unmanaged:
+		fmt.Println("Serverless is Unmanaged, Kserve will default to RawDeployment")
```

Comment:
AFAIK, `Unmanaged` would mean that the users are bringing their own Serverless setup. Right, @VaishnaviHire @zdtsw ?

If so, both `Managed` and `Unmanaged` should operate identically wrt the logic around `defaultDeploymentMode`.